packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+
+        final _TextTreeConfiguration childConfig = _childTextConfiguration(child, config);
+        if (i == children.length - 1) {
+          final String lastChildPrefixLineOne =
+              '$prefixChildren${childConfig.prefixLastChildLineOne}';
+          if (child == null) {
+            builder.writeRawLine('$lastChildPrefixLineOne<null>');
+            continue;
+          }
+          builder.writeRawLine(child.toStringDeep(
+              lastChildPrefixLineOne,
+              '$prefixChildren${childConfig.childLinkSpace}${childConfig.prefixOtherLines}'));
+          if (childConfig.footer.isNotEmpty)
+            builder.writeRaw('$prefixChildren${childConfig.childLinkSpace}${childConfig.footer}');
+
+        } else {
+          final _TextTreeConfiguration nextChildStyle =
+              _childTextConfiguration(children[i + 1], config);
+
+          final String childPrefixLineOne = '$prefixChildren${childConfig.prefixLineOne}';
+          final String childPrefixOtherLines ='$prefixChildren${nextChildStyle.linkCharacter}${childConfig.prefixOtherLines}';
+
+          if (child == null) {
+            builder.writeRawLine('$childPrefixLineOne<null>');
+            continue;
+          }
+          builder.writeRawLine(child.toStringDeep(
+              childPrefixLineOne, childPrefixOtherLines));
+          if (childConfig.footer.isNotEmpty)
+            builder.writeRaw('$prefixChildren${nextChildStyle.linkCharacter}${childConfig.footer}');
+        }
+      }
+    }
+    return builder.toString();
+  }
+}
+
+/// Diagnostics node without any properties or children.
+class _LeafDiagnosticsValue extends DiagnosticsNode {
+  _LeafDiagnosticsValue({
+    String name,
+    String header,
+    this.object,
+    bool hidden: false,
+    bool showNull: true,
+    bool showName: true,
+    bool showSeparator: true,
+    Object defaultValue,
+    DiagnosticsTreeStyle style: DiagnosticsTreeStyle.singleLine,
+  }) : super(
+    name: name,
+    header: header != null ? header : object.toString(),
+    hidden: hidden,
+    showNull: showNull,
+    showName: showName,
+    showSeparator: showSeparator,
+    defaultValue: defaultValue,
+    style: style,
+  );
+
+  @override
+  final Object object;
+
+  @override
+  List<DiagnosticsNode> getProperties() => <DiagnosticsNode>[];
+
+  @override
+  List<DiagnosticsNode> getChildren() => <DiagnosticsNode>[];
+}
+
+class _LazyLeafDiagnosticsValue extends DiagnosticsNode {
+  _LazyLeafDiagnosticsValue({
+    String name,
+    String header,
+    Object computeValue(),
+    bool hidden: false,
+    bool showNull: true,
+    bool showName: true,
+    DiagnosticsTreeStyle style = DiagnosticsTreeStyle.singleLine,
+  }) : _computeValue = computeValue,
+  super(
+    name: name,
+    header: header,
+    style: style,
+    hidden: hidden,
+    showNull: showNull,
+    showName: showName,
+  );
+
+  @override
+  Object get object => _computeValue();
+
+  Object Function() _computeValue;
```

Comment:
Shouldn't this be final?

Also I think a typedef might be clearer. It would certainly make the `DiagnosticsNode.lazyObjectProperty` factory's signature more self-documenting.